torch/csrc/jit/compiler/include/types.h

```diff
@@ -73,6 +73,7 @@ inline Dtype BinaryOpDtype(Dtype op1_dtype, Dtype op2_dtype) {
     return op1_dtype;
   }
   LOG(FATAL) << "Invalid dtypes: " << op1_dtype << ", " << op2_dtype;
+  assert_unreachable("Invalid dtypes");
```

Comment:
Interesting. We will see how to specialize LOG(FATAL) and marks that as noreturn.

Reply:
yeah I think glog just handles this correctly, we should probably switch to whatever logging/error handling facilities are already baked into pytorch.